cmd/loop/main.go

```diff
@@ -170,9 +169,28 @@ func extractPathArgs(ctx *cli.Context) (string, string, error) {
 	}
 
 	// We'll now fetch the loopdir so we can make a decision on how to
-	// properly read the cert. This will either be the default, or will have
-	// been overwritten by the end user.
+	// properly read the macaroons (if needed) and also the cert. This will
+	// either be the default, or will have been overwritten by the end
+	// user.
 	loopDir := lncfg.CleanAndExpandPath(ctx.GlobalString(loopDirFlag.Name))
+
+	// If the macaroon path as been manually provided, then we'll only
+	// target the specified file.
+	var macPath string
+	if ctx.GlobalIsSet(macaroonPathFlag.Name) {
+		macPath = lncfg.CleanAndExpandPath(ctx.GlobalString(
+			macaroonPathFlag.Name,
+		))
+
+	} else {
+		// Otherwise, we'll go into the path:
+		// loopdir/<network> in order to fetch the macaroon that we
+		// need.
+		macPath = filepath.Join(
```

Comment:
This logic seems to be slightly different from what happens from tls path. I'd think these are two file paths and should be handled exactly the same? Perhaps even via a helper function.